### DISCUSSION TITLE
Proxy authorization

### DISCUSSION THREAD

**Discussion by NekkiNeks** [minbrowser/min#2545] [Ideas]
It's great that min now has proxy settings, but the problem is that if you specify a proxy address for which authorization is required, the authorization dialog box will appear every time you log in to the browser. It is not difficult for me to remember the login and password from the proxy server, but I would like not to waste time on this procedure every time I start the browser. I would like to offer the possibility to specify the proxy address and authorization data as well. Or at least to be able to specify authorization data directly in the proxy address line in the format `user:password@example.proxy:port`, as it happens in vscode for example, which by the way also uses electron.

**Comment by PalmerAL** [comment]
Do you have any additional examples of how other applications handle this?

It seems like it wouldn't be very hard to copy what VS code is doing (links to their implementation: [1](https://github.com/microsoft/vscode/blob/a3261eae42b18ff8cb685cf649c27c8b27985eec/src/vs/platform/native/electron-main/auth.ts#L160), [2](https://github.com/microsoft/vscode/blame/a3261eae42b18ff8cb685cf649c27c8b27985eec/src/vs/platform/windows/electron-main/windowImpl.ts#L930)), but I'm wondering why this isn't standardized and if there's a standard here we should be using. Are the credentials often sensitive enough that storing them as part of the URL, as opposed to the OS keychain, would be a problem?

**Reply by NekkiNeks** [reply]
Unfortunately no, I have no other examples of this approach being used in other applications. It's just that I often use proxies, and usually all proxies involve authorization. In min I only saw the option to enter the proxy address, and my first thought was to try to do it the same way as in vscode, but it didn't work. I don't know why in vscode credentials are stored in the address string. But I can tell you that using browser level proxies is quite a popular thing, as it allows you to proxy ONLY the traffic that goes to the desired websites. Often the use of proxies implies a reduction in overall download speeds, and using a system-level proxy is not always a productive solution.  

Sry for my bad eng btw, im not a native speaker
